Exception in configure.tasks.is_task_finished

@gshefer is currently blocked by this exception, being fixed as part of SSA test refactoring.

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
File "/home/gshefer/workspace/integration_tests/cfme/containers/image.py", line 57
in perform_smartstate_analysis
    'vm', delay=5, timeout=ssa_timeout)
File "/home/gshefer/workspace/integration_tests/cfme/configure/tasks.py", line 149
in wait_analysis_finished
    delay=delay, timeout=timeout, fail_func=requests.reload)
File "/home/gshefer/workspace/integration_tests/lib/python2.7/site-packages/wait_for/__init__.py", line 145
in wait_for
    out = func(*func_args, **func_kwargs)
File "/home/gshefer/workspace/integration_tests/cfme/configure/tasks.py", line 148
in <lambda>
    wait_for(lambda: is_analysis_finished(task_name, task_type),
File "/home/gshefer/workspace/integration_tests/cfme/configure/tasks.py", line 143
in is_analysis_finished
    clear_tasks_after_success=clear_tasks_after_success)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 
tab_destination = 'AllTasks', task_name = 'Container image analysis'
expected_status = 'finished', clear_tasks_after_success = True
 
    def is_task_finished(tab_destination, task_name, expected_status, clear_tasks_after_success=True):
        view = navigate_to(Tasks, tab_destination)
>       row = view.tabs.tab_destination.table.row(task_name=task_name, state=expected_status)
E       AttributeError: 'tabs' object has no attribute 'tab_destination'

--- a/cfme/common/vm.py
+++ b/cfme/common/vm.py
@@ -748,7 +748,7 @@ class VM(BaseVM):
         sec_loc_template = "//div[@id='all_sections_treebox']//li[contains(@id, 'group_{}')]" \
                            "//span[contains(@class, 'dynatree-checkbox')]"
         sec_checkbox_loc = "//div[@id='all_sections_treebox']//li[contains(@id, 'group_{}')]" \
-                           "//span[contains(@class, 'dynatree-checkbox')]".format(sec_loc_map[section])
+            "//span[contains(@class, 'dynatree-checkbox')]".format(sec_loc_map[section])
         sec_apply_btn = "//div[@id='accordion']/a[contains(normalize-space(text()), 'Apply')]"
 
         # Deselect other sections

--- a/cfme/common/vm.py
+++ b/cfme/common/vm.py
@@ -43,10 +43,7 @@ set_ownership_form = Form(fields=[
     ('cancel_button', form_buttons.cancel)
 ])
 
-drift_table = CheckboxTable({
-    version.LOWEST: "//table[@class='style3']",
-    "5.4": "//th[normalize-space(.)='Timestamp']/ancestor::table[1]"
-})
+drift_table = CheckboxTable("//th[normalize-space(.)='Timestamp']/ancestor::table[1]")
 
 
 class _TemplateMixin(object):
@@ -709,13 +706,12 @@ class VM(BaseVM):
                 fill(self.retire_form.warn, warn)
             sel.click(form_buttons.save)
 
-
     def equal_drift_results(self, row_text, section, *indexes):
         """ Compares drift analysis results of a row specified by it's title text
 
         Args:
             row_text: Title text of the row to compare
-            section: Accordion section where the change happened; this section must be activated
+            section: Accordion section where the change happened; this section will be activated
             indexes: Indexes of results to compare starting with 0 for first row (latest result).
                      Compares all available drifts, if left empty (default).
 
@@ -767,8 +763,8 @@ class VM(BaseVM):
 
         if not tb.is_active("All attributes"):
             tb.select("All attributes")
-        d_grid = DriftGrid()
-        if any(d_grid.cell_indicates_change(row_text, i) for i in range(0, len(indexes))):
+        drift_grid = DriftGrid()
+        if any(drift_grid.cell_indicates_change(row_text, i) for i in range(0, len(indexes))):
             return False
         return True
 

--- a/cfme/common/vm.py
+++ b/cfme/common/vm.py
@@ -11,7 +11,10 @@ from cfme.exceptions import (
 from cfme.fixtures import pytest_selenium as sel
 from cfme.web_ui import (
     AngularCalendarInput, AngularSelect, Form, InfoBlock, Input, Quadicon, Select, fill, flash,
-    form_buttons, paginator, toolbar, PagedTable, SplitPagedTable, search)
+    form_buttons, paginator, toolbar, PagedTable, SplitPagedTable, search, CheckboxTable,
+    DriftGrid
+)
+import cfme.web_ui.toolbar as tb
 from utils import version
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigate_to
@@ -39,6 +42,11 @@ set_ownership_form = Form(fields=[
     ('reset_button', form_buttons.reset),
     ('cancel_button', form_buttons.cancel)
 ])
+
+drift_table = CheckboxTable({
+    version.LOWEST: "//table[@class='style3']",
+    "5.4": "//th[normalize-space(.)='Timestamp']/ancestor::table[1]"
+})
 
 
 class _TemplateMixin(object):
@@ -700,6 +708,69 @@ class VM(BaseVM):
             if warn is not None:
                 fill(self.retire_form.warn, warn)
             sel.click(form_buttons.save)
+
+
+    def equal_drift_results(self, row_text, section, *indexes):
+        """ Compares drift analysis results of a row specified by it's title text
+
+        Args:
+            row_text: Title text of the row to compare
+            section: Accordion section where the change happened; this section must be activated
+            indexes: Indexes of results to compare starting with 0 for first row (latest result).
+                     Compares all available drifts, if left empty (default).
+
+        Note:
+            There have to be at least 2 drift results available for this to work.
+
+        Returns:
+            ``True`` if equal, ``False`` otherwise.
+        """
+        # mark by indexes or mark all
+        self.load_details(refresh=True)
+        sel.click(InfoBlock("Properties", "Drift History"))
+        if indexes:
+            drift_table.select_rows_by_indexes(*indexes)
+        else:
+            # We can't compare more than 10 drift results at once
+            # so when selecting all, we have to limit it to the latest 10
+            if len(list(drift_table.rows())) > 10:
+                drift_table.select_rows_by_indexes(*range(0, min(10, len)))
+            else:
+                drift_table.select_all()
+        tb.select("Select up to 10 timestamps for Drift Analysis")
+
+        # Make sure the section we need is active/open
+        sec_loc_map = {
+            'Properties': 'Properties',
+            'Security': 'Security',
+            'Configuration': 'Configuration',
+            'My Company Tags': 'Categories'}
+        sec_loc_template = "//div[@id='all_sections_treebox']//li[contains(@id, 'group_{}')]" \
+                           "//span[contains(@class, 'dynatree-checkbox')]"
+        sec_checkbox_loc = "//div[@id='all_sections_treebox']//li[contains(@id, 'group_{}')]" \
+                           "//span[contains(@class, 'dynatree-checkbox')]".format(sec_loc_map[section])
+        sec_apply_btn = "//div[@id='accordion']/a[contains(normalize-space(text()), 'Apply')]"
+
+        # Deselect other sections
+        for other_section in sec_loc_map.keys():
+            other_section_loc = sec_loc_template.format(sec_loc_map[other_section])
+            other_section_classes = sel.get_attribute(other_section_loc + '/..', "class")
+            if other_section != section and 'dynatree-partsel' in other_section_classes:
+                # Element needs to be checked out if it has no dynatree-selected
+                if 'dynatree-selected' not in other_section_classes:
+                    sel.click(other_section_loc)
+                sel.click(other_section_loc)
+
+        # Activate the required section
+        sel.click(sec_checkbox_loc)
+        sel.click(sec_apply_btn)
+
+        if not tb.is_active("All attributes"):
+            tb.select("All attributes")
+        d_grid = DriftGrid()
+        if any(d_grid.cell_indicates_change(row_text, i) for i in range(0, len(indexes))):
+            return False
+        return True
 
 
 class Template(BaseVM, _TemplateMixin):

--- a/cfme/configure/tasks.py
+++ b/cfme/configure/tasks.py
@@ -106,7 +106,7 @@ def is_task_finished(destination, task_name, expected_status, clear_tasks_after_
         return False
 
     # throw exception if error in message
-    if 'error' in row.message.lower():
+    if 'error' in row.message.text.lower():
         raise Exception("Task {} error".format(task_name))
 
     if clear_tasks_after_success:

--- a/cfme/configure/tasks.py
+++ b/cfme/configure/tasks.py
@@ -118,7 +118,7 @@ def is_analysis_finished(name, task_type='vm', clear_tasks_after_success=True):
     tabs_data = {
         'vm': {
             'tab': 'AllTasks',
-            'task': '{}',
+            'task': 'Scan from Vm {}',
             'state': 'finished'
         },
         'host': {
@@ -136,7 +136,6 @@ def is_analysis_finished(name, task_type='vm', clear_tasks_after_success=True):
             'task': 'SmartState Analysis for [{}]',
             'state': "Finished"}
     }[task_type]
-
     return is_task_finished(tab_destination=tabs_data['tab'],
                             task_name=tabs_data['task'].format(name),
                             expected_status=tabs_data['state'],

--- a/cfme/configure/tasks.py
+++ b/cfme/configure/tasks.py
@@ -14,6 +14,7 @@ import cfme.fixtures.pytest_selenium as sel
 from cfme.web_ui import Form, Region, CheckboxTable, fill, match_location
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+from utils.log import logger
 from utils.wait import wait_for, TimedOutError
 from cfme.services import requests
 
@@ -95,15 +96,18 @@ def is_cluster_analysis_finished(name, **kwargs):
     return is_analysis_finished(name=name, task_type='cluster', **kwargs)
 
 
-def is_task_finished(tab_destination, task_name, expected_status, clear_tasks_after_success=True):
-    view = navigate_to(Tasks, tab_destination)
-    row = view.tabs.tab_destination.table.row(task_name=task_name, state=expected_status)
-    if row is None:
+def is_task_finished(destination, task_name, expected_status, clear_tasks_after_success=True):
+    view = navigate_to(Tasks, destination)
+    tab_view = getattr(view.tabs, destination.lower())
+    try:
+        row = tab_view.table.row(task_name=task_name, state=expected_status)
+    except IndexError:
+        logger.warn('IndexError exception suppressed when searching for task row, no match found.')
         return False
 
-    # throw exception if status is error
-    if 'Error' in row.title:
-        raise Exception("Task {} errored".format(task_name))
+    # throw exception if error in message
+    if 'error' in row.message.lower():
+        raise Exception("Task {} error".format(task_name))
 
     if clear_tasks_after_success:
         # Remove all finished tasks so they wouldn't poison other tests
@@ -136,7 +140,7 @@ def is_analysis_finished(name, task_type='vm', clear_tasks_after_success=True):
             'task': 'SmartState Analysis for [{}]',
             'state': "Finished"}
     }[task_type]
-    return is_task_finished(tab_destination=tabs_data['tab'],
+    return is_task_finished(destination=tabs_data['tab'],
                             task_name=tabs_data['task'].format(name),
                             expected_status=tabs_data['state'],
                             clear_tasks_after_success=clear_tasks_after_success)

--- a/cfme/configure/tasks.py
+++ b/cfme/configure/tasks.py
@@ -111,7 +111,7 @@ def is_task_finished(destination, task_name, expected_status, clear_tasks_after_
 
     if clear_tasks_after_success:
         # Remove all finished tasks so they wouldn't poison other tests
-        view.delete('Delete All', handle_alert=True)
+        view.delete.item_select('Delete All', handle_alert=True)
 
     return True
 

--- a/cfme/infrastructure/datastore.py
+++ b/cfme/infrastructure/datastore.py
@@ -85,12 +85,8 @@ class Datastore(Pretty, Navigatable):
              message="Wait datastore to appear", num_sec=1000, fail_func=sel.refresh)
 
     def load_details(self):
-        # todo: to remove this context related functionality along with making provider
-        # param mandatory
-        if not self.provider:
-            navigate_to(self, 'Details')
-        else:
-            navigate_to(self, 'DetailsFromProvider')
+        # todo: to remove this context related functionality
+        navigate_to(self, 'Details')
 
     def get_detail(self, *ident):
         """ Gets details from the details infoblock
@@ -110,10 +106,7 @@ class Datastore(Pretty, Navigatable):
         Returns: List of strings with names or `[]` if no hosts found.
         """
         self.load_details()
-        try:
-            sel.click(details_page.infoblock.element("Relationships", "Hosts"))
-        except sel.NoSuchElementException:
-            sel.click(InfoBlock('Relationships', 'Hosts'))
+        sel.click(InfoBlock('Relationships', 'Hosts'))
         return [q.name for q in Quadicon.all("host")]
 
     def get_vms(self):
@@ -211,8 +204,10 @@ class Details(CFMENavigateStep):
 
 @navigator.register(Datastore, 'DetailsFromProvider')
 class DetailsFromProvider(CFMENavigateStep):
-    def step(self):
+    def prerequisite(self, *args, **kwargs):
         navigate_to(self.obj.provider, 'Details')
+
+    def step(self):
         list_acc.select('Relationships', 'Datastores', by_title=False, partial=True)
         sel.click(Quadicon(self.obj.name, self.obj.quad_name))
 

--- a/cfme/infrastructure/host.py
+++ b/cfme/infrastructure/host.py
@@ -452,7 +452,7 @@ class All(CFMENavigateStep):
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', '/host')
+        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Hosts')
 
     def resetter(self):
         tb.select("Grid View")

--- a/cfme/infrastructure/provider/virtualcenter.py
+++ b/cfme/infrastructure/provider/virtualcenter.py
@@ -26,9 +26,14 @@ class VMwareProvider(InfraProvider):
 
     def deployment_helper(self, deploy_args):
         """ Used in utils.virtual_machines """
+        # Called within a dictionary update. Since we want to remove key/value pairs, return the
+        # entire dictionary
+        del deploy_args['username']
+        del deploy_args['password']
         if "allowed_datastores" not in deploy_args and "allowed_datastores" in self.data:
-            return {'allowed_datastores': self.data['allowed_datastores']}
-        return {}
+            deploy_args['allowed_datastores'] = self.data['allowed_datastores']
+
+        return deploy_args
 
     @classmethod
     def from_config(cls, prov_config, prov_key, appliance=None):

--- a/cfme/infrastructure/virtual_machines.py
+++ b/cfme/infrastructure/virtual_machines.py
@@ -18,8 +18,8 @@ from cfme.services import requests
 import cfme.web_ui.toolbar as tb
 from cfme.web_ui import (
     CheckboxTree, Form, InfoBlock, Region, Quadicon, Tree, accordion, fill, flash, form_buttons,
-    paginator, toolbar, Calendar, Select, Input, CheckboxTable, DriftGrid, match_location,
-    BootstrapTreeview, summary_title, Table, search
+    match_location, Table, search,paginator, toolbar, Calendar, Select, Input, CheckboxTable,
+    summary_title, BootstrapTreeview
 )
 from cfme.web_ui.search import search_box
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
@@ -77,6 +77,9 @@ retirement_date_form = Form(fields=[
 retire_remove_button = "//span[@id='remove_button']/a/img"
 
 match_page = partial(match_location, controller='vm_infra', title='Virtual Machines')
+vm_templates_tree = partial(accordion.tree, "VMs & Templates")
+vms_tree = partial(accordion.tree, "VMs")
+templates_tree = partial(accordion.tree, "Templates")
 
 
 def reset_page():
@@ -340,68 +343,6 @@ class Vm(BaseVM):
 
     def get_collection_via_rest(self):
         return self.appliance.rest_api.collections.vms
-
-    def equal_drift_results(self, row_text, section, *indexes):
-        """ Compares drift analysis results of a row specified by it's title text
-
-        Args:
-            row_text: Title text of the row to compare
-            section: Accordion section where the change happened; this section must be activated
-            indexes: Indexes of results to compare starting with 0 for first row (latest result).
-                     Compares all available drifts, if left empty (default).
-
-        Note:
-            There have to be at least 2 drift results available for this to work.
-
-        Returns:
-            ``True`` if equal, ``False`` otherwise.
-        """
-        # mark by indexes or mark all
-        self.load_details(refresh=True)
-        sel.click(InfoBlock("Properties", "Drift History"))
-        if indexes:
-            drift_table.select_rows_by_indexes(*indexes)
-        else:
-            # We can't compare more than 10 drift results at once
-            # so when selecting all, we have to limit it to the latest 10
-            if len(list(drift_table.rows())) > 10:
-                drift_table.select_rows_by_indexes(*range(0, min(10, len)))
-            else:
-                drift_table.select_all()
-        tb.select("Select up to 10 timestamps for Drift Analysis")
-
-        # Make sure the section we need is active/open
-        sec_loc_map = {
-            'Properties': 'Properties',
-            'Security': 'Security',
-            'Configuration': 'Configuration',
-            'My Company Tags': 'Categories'}
-        sec_loc_template = "//div[@id='all_sections_treebox']//li[contains(@id, 'group_{}')]"\
-            "//span[contains(@class, 'dynatree-checkbox')]"
-        sec_checkbox_loc = "//div[@id='all_sections_treebox']//li[contains(@id, 'group_{}')]"\
-            "//span[contains(@class, 'dynatree-checkbox')]".format(sec_loc_map[section])
-        sec_apply_btn = "//div[@id='accordion']/a[contains(normalize-space(text()), 'Apply')]"
-
-        # Deselect other sections
-        for other_section in sec_loc_map.keys():
-            other_section_loc = sec_loc_template.format(sec_loc_map[other_section])
-            other_section_classes = sel.get_attribute(other_section_loc + '/..', "class")
-            if other_section != section and 'dynatree-partsel' in other_section_classes:
-                # Element needs to be checked out if it has no dynatree-selected
-                if 'dynatree-selected' not in other_section_classes:
-                    sel.click(other_section_loc)
-                sel.click(other_section_loc)
-
-        # Activate the required section
-        sel.click(sec_checkbox_loc)
-        sel.click(sec_apply_btn)
-
-        if not tb.is_active("All attributes"):
-            tb.select("All attributes")
-        d_grid = DriftGrid()
-        if any(d_grid.cell_indicates_change(row_text, i) for i in range(0, len(indexes))):
-            return False
-        return True
 
     @property
     def cluster_id(self):

--- a/cfme/infrastructure/virtual_machines.py
+++ b/cfme/infrastructure/virtual_machines.py
@@ -18,8 +18,8 @@ from cfme.services import requests
 import cfme.web_ui.toolbar as tb
 from cfme.web_ui import (
     CheckboxTree, Form, InfoBlock, Region, Quadicon, Tree, accordion, fill, flash, form_buttons,
-    match_location, Table, search,paginator, toolbar, Calendar, Select, Input, CheckboxTable,
-    summary_title, BootstrapTreeview
+    match_location, Table, search, paginator, toolbar, Calendar, Select, Input, CheckboxTable,
+    summary_title, BootstrapTreeview, AngularSelect
 )
 from cfme.web_ui.search import search_box
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
@@ -353,7 +353,9 @@ class Vm(BaseVM):
     class CfmeRelationship(object):
         relationship_form = Form(
             fields=[
-                ('server_select', Select("//*[@id='server_id']")),
+                ('server_select', {
+                    version.LOWEST: Select("//*[@id='server_id']"),
+                    "5.5": AngularSelect("server_id")}),
                 ('save_button', form_buttons.save),
                 ('reset_button', form_buttons.reset),
                 ('cancel_button', form_buttons.cancel)
@@ -383,17 +385,9 @@ class Vm(BaseVM):
                 fill(self.relationship_form, {'server_select': option},
                      action=self.relationship_form.cancel_button)
             else:
-                fill(self.relationship_form, {'server_select': option},
-                     action=self.relationship_form.save_button)
-                # something weird going on where changing the select doesn't POST to undim save
-                sel.wait_for_ajax()
-                if self.relationship_form.save_button.is_dimmed:
-                    logger.warning("Worked around dimmed save button")
-                    sel.browser().execute_script(
-                        "$j.ajax({type: 'POST', url: '/vm_infra/evm_relationship_field_changed',"
-                        " data: {'server_id':'%s'}})" % (server_id))
-                    sel.click(form_buttons.FormButton(
-                        "Save Changes", dimmed_alt="Save", force_click=True))
+                fill(self.relationship_form, {'server_select': option})
+                sel.click(form_buttons.FormButton(
+                    "Save Changes", dimmed_alt="Save", force_click=True))
                 flash.assert_success_message("Management Engine Relationship saved")
 
 

--- a/cfme/infrastructure/virtual_machines.py
+++ b/cfme/infrastructure/virtual_machines.py
@@ -353,9 +353,7 @@ class Vm(BaseVM):
     class CfmeRelationship(object):
         relationship_form = Form(
             fields=[
-                ('server_select', {
-                    version.LOWEST: Select("//*[@id='server_id']"),
-                    "5.5": AngularSelect("server_id")}),
+                ('server_select', AngularSelect("server_id")),
                 ('save_button', form_buttons.save),
                 ('reset_button', form_buttons.reset),
                 ('cancel_button', form_buttons.cancel)

--- a/cfme/tests/infrastructure/test_instance_analysis.py
+++ b/cfme/tests/infrastructure/test_instance_analysis.py
@@ -14,18 +14,15 @@ from cfme.control.explorer.policies import VMControlPolicy
 from cfme.control.explorer.actions import Action
 from cfme.fixtures import pytest_selenium as sel
 from cfme.infrastructure import host, datastore
-from cfme.provisioning import do_vm_provisioning
 from cfme.web_ui import InfoBlock, DriftGrid, toolbar
 from fixtures.pytest_store import store
 from utils import testgen, ssh, safe_string, version, error
 from utils.conf import cfme_data
 from utils.log import logger
 from utils.wait import wait_for
-from utils.blockers import GH, BZ
+from utils.blockers import BZ
 
-pytestmark = [pytest.mark.meta(blockers=["GH#ManageIQ/manageiq:6939"],
-                               unblock=lambda provider: provider.type != 'rhevm'),
-              pytest.mark.tier(3), test_requirements.smartstate,
+pytestmark = [pytest.mark.tier(3), test_requirements.smartstate,
               pytest.mark.meta(blockers=[1378447])]
 
 WINDOWS = {'id': "Red Hat Enterprise Windows", 'icon': 'windows'}
@@ -35,22 +32,30 @@ RPM_BASED = {
         'id': "Red Hat", 'release-file': '/etc/redhat-release', 'icon': 'linux_redhat',
         'package': "kernel", 'install-command': "",  # We don't install stuff on RHEL
         'package-number': 'rpm -qa | wc -l',
-        'services-number': 'systemctl -a --type service -o cat --no-legend --no-pager | wc -l'},
+        'services-number': 'echo $((`ls -lL /etc/init.d | egrep -i -v "readme|total" | wc -l` + '
+                           '`ls -l /usr/lib/systemd/system | grep service | wc -l` + '
+                           '`ls -l /usr/lib/systemd/user | grep service | wc -l`))'},
     'centos': {
         'id': "CentOS", 'release-file': '/etc/centos-release', 'icon': 'linux_centos',
         'package': 'iso-codes', 'install-command': 'yum install -y {}',
         'package-number': 'rpm -qa | wc -l',
-        'services-number': 'systemctl -a --type service -o cat --no-legend --no-pager | wc -l'},
+        'services-number': 'echo $((`ls -lL /etc/init.d | egrep -i -v "readme|total" | wc -l` +'
+                           ' `ls -l /usr/lib/systemd/system | grep service | wc -l` +'
+                           ' `ls -l /usr/lib/systemd/user | grep service | wc -l`))'},
     'fedora': {
         'id': 'Fedora', 'release-file': '/etc/fedora-release', 'icon': 'linux_fedora',
         'package': 'iso-codes', 'install-command': 'dnf install -y {}',
         'package-number': 'rpm -qa | wc -l',
-        'services-number': 'systemctl -a --type service -o cat --no-legend --no-pager | wc -l'},
+        'services-number': 'echo $((`ls -lL /etc/init.d | egrep -i -v "readme|total" | wc -l` +'
+                           ' `ls -l /usr/lib/systemd/system | grep service | wc -l` +'
+                           ' `ls -l /usr/lib/systemd/user | grep service | wc -l`))'},
     'suse': {
         'id': 'Suse', 'release-file': '/etc/SuSE-release', 'icon': 'linux_suse',
         'package': 'iso-codes', 'install-command': 'zypper install -y {}',
         'package-number': 'rpm -qa | wc -l',
-        'services-number': 'systemctl -a --type service -o cat --no-legend --no-pager | wc -l'},
+        'services-number': 'echo $((`ls -lL /etc/init.d | egrep -i -v "readme|total" | wc -l` +'
+                           ' `ls -l /usr/lib/systemd/system | grep service | wc -l` +'
+                           ' `ls -l /usr/lib/systemd/user | grep service | wc -l`))'},
 }
 
 DEB_BASED = {
@@ -59,13 +64,15 @@ DEB_BASED = {
         'package': 'iso-codes',
         'install-command': 'env DEBIAN_FRONTEND=noninteractive apt-get -y install {}',
         'package-number': "dpkg --get-selections | wc -l",
-        'services-number': 'chkconfig --list | wc -l'},
+        'services-number': 'echo $((`ls -alL /etc/init.d | egrep -iv "readme|total|drwx" | wc -l` +'
+                           ' `ls -alL /etc/systemd/system/ | grep service | wc -l`))'},
     'debian': {
         'id': 'Debian ', 'release-file': '/etc/issue.net', 'icon': 'linux_debian',
         'package': 'iso-codes',
         'install-command': 'env DEBIAN_FRONTEND=noninteractive apt-get -y install {}',
         'package-number': 'dpkg --get-selections | wc -l',
-        'services-number': 'chkconfig --list | wc -l'},
+        'services-number': 'echo $((`ls -alL /etc/init.d | egrep -iv "readme|total|drwx" | wc -l` +'
+                           ' `ls -alL /etc/systemd/system/ | grep service | wc -l`))'},
 }
 
 ssa_expect_file = "/etc/hosts"
@@ -122,12 +129,8 @@ def pytest_generate_tests(metafunc):
 
 @pytest.fixture(scope="module")
 def local_setup_provider(request, setup_provider_modscope, provider, vm_analysis_data):
-    if provider.type == 'rhevm' and version.current_version() < "5.5":
-        # See https://bugzilla.redhat.com/show_bug.cgi?id=1300030
-        pytest.skip("SSA is not supported on RHEVM for appliances earlier than 5.5 and upstream")
-    if GH("ManageIQ/manageiq:6506").blocks:
-        pytest.skip("Upstream provisioning is blocked by" +
-                    "https://github.com/ManageIQ/manageiq/issues/6506")
+
+    # TODO: allow for vddk parameterization
     if provider.type == 'virtualcenter':
         store.current_appliance.install_vddk(reboot=True, wait_for_web_ui_after_reboot=True)
         store.current_appliance.browser.quit_browser()
@@ -150,6 +153,7 @@ def set_host_credentials(request, provider, vm_analysis_data):
     host_list = cfme_data.get('management_systems', {})[provider.key].get('hosts', [])
     host_data = [x for x in host_list if x.name == vm_analysis_data['host']][0]
 
+    # has valid creds appears broken
     if not test_host.has_valid_credentials:
         test_host.update(
             updates={'credentials': host.get_credentials_from_config(host_data['credentials'])},
@@ -168,68 +172,65 @@ def set_host_credentials(request, provider, vm_analysis_data):
 
 @pytest.fixture(scope="module")
 def vm_name(provider, analysis_type):
-    vm_name = 'test_ssa_{}-{}'.format(fauxfactory.gen_alphanumeric(), analysis_type)
+    vm_name = 'test-ssa-{}-{}'.format(fauxfactory.gen_alphanumeric(), analysis_type)
     return vm_name
 
 
 @pytest.fixture(scope="module")
 def vm_analysis_data(provider, analysis_type):
-    base_data = provider.data.get('vm_analysis_new', {}).get('provisioning', {})
-    base_data.update(provider.data.get('vm_analysis_new', {}).get('vms', {}).get(analysis_type, {}))
-    return base_data
+
+    pdata = provider.data
+    provisioning_data = pdata.get('vm_analysis_new', {}).get('provisioning', {})
+
+    # Setup up the provisioning data for the instance/vm
+    # Will default to provisioning items under vm_analysis but if not defined
+    #   falls back to items under provider['provisioning'] key
+    if 'host' not in provisioning_data and 'host' in pdata['provisioning']:
+        provisioning_data['host'] = pdata['provisioning']['host']
+    if 'datastore' not in provisioning_data and 'datastore' in pdata['provisioning']:
+        provisioning_data['datastore'] = pdata['provisioning']['datastore']
+    if 'vlan' not in provisioning_data and 'vlan' in pdata['provisioning']:
+        provisioning_data['vlan'] = pdata['provisioning']['vlan']
+    if 'instance_type' not in provisioning_data and 'instance_type' in pdata['provisioning']:
+        provisioning_data['instance_type'] = pdata['provisioning']['instance_type']
+    if 'availability_zone' not in provisioning_data and \
+            'availability_zone' in pdata['provisioning']:
+        provisioning_data['availability_zone'] = pdata['provisioning']['availability_zone']
+    if 'security_group' not in provisioning_data and 'security_group' in pdata['provisioning']:
+        provisioning_data['security_group'] = pdata['provisioning']['security_group']
+    if 'cloud_network' not in provisioning_data and 'cloud_network' in pdata['provisioning']:
+        provisioning_data['cloud_network'] = pdata['provisioning']['cloud_network']
+
+    # If defined, tries to find cluster from provisioning, then provider definition itself
+    if provider.type == 'rhevm':
+        if 'cluster' not in provisioning_data:
+            if 'cluster' not in pdata['provisioning']:
+                provisioning_data['cluster'] = pdata['default_cluster']
+            else:
+                provisioning_data['cluster'] = pdata['provisioning']['cluster']
+
+    provisioning_data.update(
+        provider.data.get('vm_analysis_new', {}).get('vms', {}).get(analysis_type, {}))
+    return provisioning_data
 
 
 @pytest.fixture(scope="module")
 def instance(request, local_setup_provider, provider, vm_name, vm_analysis_data):
     """ Fixture to provision instance on the provider """
 
-    template = vm_analysis_data.get('image', None)
-    host_name, datastore_name = map(vm_analysis_data.get, ('host', 'datastore'))
+    vm = VM.factory(vm_name, provider, template_name=vm_analysis_data['image'])
+    request.addfinalizer(lambda: cleanup_vm(vm_name, provider))
 
-    mgmt_system = provider.get_mgmt_system()
+    provision_data = vm_analysis_data.copy()
+    del provision_data['image']
+    vm.create_on_provider(find_in_cfme=True, **provision_data)
 
-    provisioning_data = {
-        'vm_name': vm_name,
-        'host_name': {'name': [host_name]},
-        'datastore_name': {'name': [datastore_name]},
-    }
-
-    try:
-        provisioning_data['vlan'] = vm_analysis_data['vlan']
-    except KeyError:
-        # provisioning['vlan'] is required for rhevm provisioning
-        if provider.type == 'rhevm':
-            raise pytest.fail('rhevm requires a vlan value in provisioning info')
-
-    vm = VM.factory(vm_name, provider)
-
-    connect_ip = None
     if provider.type == "openstack":
-        image = vm_analysis_data['image']
-        vm = VM.factory(vm_name, provider, image)
-        request.addfinalizer(vm.delete_from_provider)
-        connect_ip = mgmt_system.get_first_floating_ip()
-        provider.refresh_provider_relationships(method='ui')
-        inst_args = {
-            'email': 'image_provisioner@example.com',
-            'first_name': 'Image',
-            'last_name': 'Provisioner',
-            'template_name': image,
-            'notes': ('Testing provisioning from image {} to vm {} on provider {}'.format(
-                image, vm_name, provider.key)),
-            'instance_type': vm_analysis_data['instance_type'],
-            'availability_zone': vm_analysis_data['availability_zone'],
-            'security_groups': [vm_analysis_data['security_group']],
-            'cloud_network': vm_analysis_data['cloud_network'],
-            'public_ip_address': connect_ip,
-        }
-        vm.create(**inst_args)
-    else:
-        request.addfinalizer(lambda: cleanup_vm(vm_name, provider))
-        do_vm_provisioning(template, provider, vm_name, provisioning_data, request, None,
-                           num_sec=6000)
+        vm.provider.mgmt.assign_floating_ip(vm.name, 'public')
+
     logger.info("VM %s provisioned, waiting for IP address to be assigned", vm_name)
 
+    mgmt_system = provider.get_mgmt_system()
     @pytest.wait_for(timeout="20m", delay=5)
     def get_ip_address():
         logger.info("Power state for {} vm: {}, is_vm_stopped: {}".format(
@@ -251,14 +252,18 @@ def instance(request, local_setup_provider, provider, vm_name, vm_analysis_data)
         logger.info("Waiting for %s to be available via SSH", connect_ip)
         ssh_client = ssh.SSHClient(hostname=connect_ip, username=vm_analysis_data['username'],
                                    password=vm_analysis_data['password'], port=22)
-        wait_for(ssh_client.uptime, num_sec=3600, handle_exception=False)
+        wait_for(ssh_client.uptime, num_sec=3600, handle_exception=True)
         vm.ssh = ssh_client
-
     vm.system_type = detect_system_type(vm)
     logger.info("Detected system type: %s", vm.system_type)
     vm.image = vm_analysis_data['image']
     vm.connect_ip = connect_ip
 
+    # TODO:  This is completely wrong and needs to be fixed
+    #   CFME relationship is suppose to be set to the appliance, which is required
+    #   to be placed within the same datastore that the VM resides
+    #
+    #   Also, if rhev and iscsi, it need direct_lun
     if provider.type == 'rhevm':
         logger.info("Setting a relationship between VM and appliance")
         from cfme.infrastructure.virtual_machines import Vm
@@ -332,9 +337,6 @@ def detect_system_type(vm):
 @pytest.mark.tier(1)
 @pytest.mark.long_running
 @pytest.mark.meta(blockers=[
-    BZ(1311134, unblock=lambda provider: provider.type != 'rhevm'),
-    BZ(1311218, unblock=lambda provider:
-        provider.type != 'virtualcenter' or provider.version < "6"),
     BZ(1320248, unblock=lambda provider: version.current_version() >= "5.5")])
 def test_ssa_template(request, local_setup_provider, provider, soft_assert, vm_analysis_data):
     """ Tests SSA can be performed on a template
@@ -347,7 +349,7 @@ def test_ssa_template(request, local_setup_provider, provider, soft_assert, vm_a
     template = Template.factory(template_name, provider, template=True)
 
     # Set credentials to all hosts set for this datastore
-    if provider.type != 'openstack':
+    if provider.type in ['virtualcenter', 'rhevm']:
         datastore_name = vm_analysis_data['datastore']
         test_datastore = datastore.Datastore(datastore_name, provider.key)
         host_list = cfme_data.get('management_systems', {})[provider.key].get('hosts', [])
@@ -367,7 +369,7 @@ def test_ssa_template(request, local_setup_provider, provider, soft_assert, vm_a
 
     template.smartstate_scan()
     wait_for(lambda: is_vm_analysis_finished(template_name),
-             delay=15, timeout="10m", fail_func=lambda: toolbar.select('Reload'))
+             delay=15, timeout="35m", fail_func=lambda: toolbar.select('Reload'))
 
     # Check release and quadricon
     quadicon_os_icon = template.find_quadicon().os
@@ -404,6 +406,7 @@ def test_ssa_template(request, local_setup_provider, provider, soft_assert, vm_a
         soft_assert(c_fs_drivers != '0', "fs drivers: '{}' != '0'".format(c_fs_drivers))
 
 
+@pytest.mark.tier(2)
 @pytest.mark.long_running
 def test_ssa_vm(provider, instance, soft_assert):
     """ Tests SSA can be performed and returns sane results
@@ -411,6 +414,11 @@ def test_ssa_vm(provider, instance, soft_assert):
     Metadata:
         test_flag: vm_analysis
     """
+
+    # TODO: check if previously scanned?
+    #       delete the vm itself if it did have a scan already
+    #       delete all previous scan tasks
+
 
     e_users = None
     e_groups = None
@@ -431,7 +439,7 @@ def test_ssa_vm(provider, instance, soft_assert):
 
     instance.smartstate_scan()
     wait_for(lambda: is_vm_analysis_finished(instance.name),
-             delay=15, timeout="15m", fail_func=lambda: toolbar.select('Reload'))
+             delay=15, timeout="35m", fail_func=lambda: toolbar.select('Reload'))
 
     # Check release and quadricon
     quadicon_os_icon = instance.find_quadicon().os
@@ -480,14 +488,14 @@ def test_ssa_vm(provider, instance, soft_assert):
         soft_assert(c_kernel_drivers != '0', "kernel drivers: '{}' != '0'".format(c_kernel_drivers))
         soft_assert(c_fs_drivers != '0', "fs drivers: '{}' != '0'".format(c_fs_drivers))
 
-    image_label = 'Parent VM'
-    if provider.type == 'openstack':
-        image_label = 'VM Template'
+    # image_label = 'Parent VM'
+    # if provider.type == 'openstack':
+    #     image_label = 'VM Template'
     # 5.4 doesn't have Parent VM field
-    if version.current_version() > "5.5" and provider.type != 'openstack':
-        c_image = InfoBlock.text('Relationships', image_label)
-        soft_assert(c_image == instance.image,
-                    "image: '{}' != '{}'".format(c_image, instance.image))
+    # if version.current_version() > "5.5" and provider.type != 'openstack':
+    #     c_image = InfoBlock.text('Relationships', image_label)
+    #     soft_assert(c_image == instance.image,
+    #                 "image: '{}' != '{}'".format(c_image, instance.image))
 
 
 @pytest.mark.long_running
@@ -504,12 +512,14 @@ def test_ssa_users(provider, instance, soft_assert):
     # So we simply check that user list doesn't cause any Rails errors
     if instance.system_type != WINDOWS:
         # Add a new user
+        # force ssh re-connection
+        instance.ssh.close()
         instance.ssh.run_command("userdel {0} || useradd {0}".format(username))
         expected = instance.ssh.run_command("cat /etc/passwd | wc -l").output.strip('\n')
 
     instance.smartstate_scan()
     wait_for(lambda: is_vm_analysis_finished(instance.name),
-             delay=15, timeout="15m", fail_func=lambda: toolbar.select('Reload'))
+             delay=15, timeout="35m", fail_func=lambda: toolbar.select('Reload'))
 
     # Check that all data has been fetched
     current = instance.get_detail(properties=('Security', 'Users'))
@@ -535,12 +545,14 @@ def test_ssa_groups(provider, instance, soft_assert):
 
     if instance.system_type != WINDOWS:
         # Add a new group
+        # force ssh re-connection
+        instance.ssh.close()
         instance.ssh.run_command("groupdel {0} || groupadd {0}".format(group))
         expected = instance.ssh.run_command("cat /etc/group | wc -l").output.strip('\n')
 
     instance.smartstate_scan()
     wait_for(lambda: is_vm_analysis_finished(instance.name),
-             delay=15, timeout="15m", fail_func=lambda: toolbar.select('Reload'))
+             delay=15, timeout="35m", fail_func=lambda: toolbar.select('Reload'))
 
     # Check that all data has been fetched
     current = instance.get_detail(properties=('Security', 'Groups'))
@@ -574,6 +586,8 @@ def test_ssa_packages(provider, instance, soft_assert):
     package_number_command = instance.system_type['package-number']
 
     cmd = package_command.format(package_name)
+    # force ssh re-connection
+    instance.ssh.close()
     output = instance.ssh.run_command(cmd.format(package_name)).output
     logger.info("%s output:\n%s", cmd, output)
 
@@ -581,7 +595,7 @@ def test_ssa_packages(provider, instance, soft_assert):
 
     instance.smartstate_scan()
     wait_for(lambda: is_vm_analysis_finished(instance.name),
-             delay=15, timeout="15m", fail_func=lambda: toolbar.select('Reload'))
+             delay=15, timeout="35m", fail_func=lambda: toolbar.select('Reload'))
 
     # Check that all data has been fetched
     current = instance.get_detail(properties=('Configuration', 'Packages'))
@@ -602,7 +616,7 @@ def test_ssa_files(provider, instance, policy_profile, soft_assert):
 
     instance.smartstate_scan()
     wait_for(lambda: is_vm_analysis_finished(instance.name),
-             delay=15, timeout="15m", fail_func=lambda: toolbar.select('Reload'))
+             delay=15, timeout="35m", fail_func=lambda: toolbar.select('Reload'))
 
     # Check that all data has been fetched
     current = instance.get_detail(properties=('Configuration', 'Files'))
@@ -613,6 +627,7 @@ def test_ssa_files(provider, instance, policy_profile, soft_assert):
         pytest.fail("File {0} was not found".format(ssa_expect_file))
 
 
+@pytest.mark.tier(2)
 @pytest.mark.long_running
 def test_drift_analysis(request, provider, instance, soft_assert):
     """ Tests drift analysis is correct
@@ -628,7 +643,7 @@ def test_drift_analysis(request, provider, instance, soft_assert):
         drift_num_orig = int(drift_orig)
     instance.smartstate_scan()
     wait_for(lambda: is_vm_analysis_finished(instance.name),
-             delay=15, timeout="15m", fail_func=lambda: toolbar.select('Reload'))
+             delay=15, timeout="35m", fail_func=lambda: toolbar.select('Reload'))
     instance.load_details()
     wait_for(
         lambda: int(InfoBlock("Relationships", "Drift History").text) == drift_num_orig + 1,
@@ -646,7 +661,7 @@ def test_drift_analysis(request, provider, instance, soft_assert):
 
     instance.smartstate_scan()
     wait_for(lambda: is_vm_analysis_finished(instance.name),
-             delay=15, timeout="15m", fail_func=lambda: toolbar.select('Reload'))
+             delay=15, timeout="35m", fail_func=lambda: toolbar.select('Reload'))
     instance.load_details()
     wait_for(
         lambda: int(InfoBlock("Relationships", "Drift History").text) == drift_new + 1,

--- a/cfme/tests/infrastructure/test_instance_analysis.py
+++ b/cfme/tests/infrastructure/test_instance_analysis.py
@@ -15,13 +15,12 @@ from cfme.control.explorer.policies import VMControlPolicy
 from cfme.control.explorer.actions import Action
 from cfme.fixtures import pytest_selenium as sel
 from cfme.infrastructure import host, datastore
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
 from cfme.web_ui import InfoBlock, DriftGrid, toolbar
-from fixtures.pytest_store import store
-from utils import testgen, ssh, safe_string, version, error
+from utils import testgen, ssh, safe_string, error
 from utils.conf import cfme_data
 from utils.log import logger
 from utils.wait import wait_for
-from utils.blockers import BZ
 
 pytestmark = [pytest.mark.tier(3), test_requirements.smartstate]
 
@@ -128,13 +127,13 @@ def pytest_generate_tests(metafunc):
 
 
 @pytest.fixture(scope="module")
-def local_setup_provider(request, setup_provider_modscope, provider, vm_analysis_data):
+def local_setup_provider(request, setup_provider_modscope, provider, vm_analysis_data, appliance):
 
     # TODO: allow for vddk parameterization
-    if provider.type == 'virtualcenter':
-        store.current_appliance.install_vddk(reboot=True, wait_for_web_ui_after_reboot=True)
-        store.current_appliance.browser.quit_browser()
-        store.current_appliance.browser.open_browser()
+    if provider.one_of(VMwareProvider):
+        appliance.install_vddk(reboot=True, wait_for_web_ui_after_reboot=True)
+        appliance.browser.quit_browser()
+        appliance.browser.open_browser()
         set_host_credentials(request, provider, vm_analysis_data)
 
     # Make sure all roles are set
@@ -192,7 +191,8 @@ def vm_analysis_data(provider, analysis_type):
         provisioning_data.setdefault('vlan', pdata['provisioning']['vlan'])
     if isinstance(provider, CloudProvider):
         provisioning_data.setdefault('instance_type', pdata['provisioning']['instance_type'])
-        provisioning_data.setdefault('availability_zone', pdata['provisioning']['availability_zone'])
+        provisioning_data.setdefault('availability_zone',
+                                     pdata['provisioning']['availability_zone'])
         provisioning_data.setdefault('security_group', pdata['provisioning']['security_group'])
         provisioning_data.setdefault('cloud_network', pdata['provisioning']['cloud_network'])
 
@@ -209,7 +209,7 @@ def vm_analysis_data(provider, analysis_type):
 
 
 @pytest.fixture(scope="module")
-def instance(request, local_setup_provider, provider, vm_name, vm_analysis_data):
+def instance(request, local_setup_provider, provider, vm_name, vm_analysis_data, appliance):
     """ Fixture to provision instance on the provider """
 
     vm = VM.factory(vm_name, provider, template_name=vm_analysis_data['image'])
@@ -225,6 +225,7 @@ def instance(request, local_setup_provider, provider, vm_name, vm_analysis_data)
     logger.info("VM %s provisioned, waiting for IP address to be assigned", vm_name)
 
     mgmt_system = provider.get_mgmt_system()
+
     @pytest.wait_for(timeout="20m", delay=5)
     def get_ip_address():
         logger.info("Power state for {} vm: {}, is_vm_stopped: {}".format(
@@ -262,7 +263,7 @@ def instance(request, local_setup_provider, provider, vm_name, vm_analysis_data)
         logger.info("Setting a relationship between VM and appliance")
         from cfme.infrastructure.virtual_machines import Vm
         cfme_rel = Vm.CfmeRelationship(vm)
-        server_name = store.current_appliance.server_name()
+        server_name = appliance.server_name()
         cfme_rel.set_relationship(str(server_name), configuration.server_id())
     return vm
 
@@ -330,8 +331,6 @@ def detect_system_type(vm):
 
 @pytest.mark.tier(1)
 @pytest.mark.long_running
-@pytest.mark.meta(blockers=[
-    BZ(1320248, unblock=lambda provider: version.current_version() >= "5.5")])
 def test_ssa_template(request, local_setup_provider, provider, soft_assert, vm_analysis_data):
     """ Tests SSA can be performed on a template
 
@@ -462,11 +461,10 @@ def test_ssa_vm(provider, instance, soft_assert):
     if instance.system_type != WINDOWS:
         soft_assert(c_users == e_users, "users: '{}' != '{}'".format(c_users, e_users))
         soft_assert(c_groups == e_groups, "groups: '{}' != '{}'".format(c_groups, e_groups))
-        soft_assert(c_packages == e_packages,
-            "packages: '{}' != '{}'".format(c_packages, e_packages))
-        if not BZ("1312971").blocks:
-            soft_assert(c_services == e_services,
-                        "services: '{}' != '{}'".format(c_services, e_services))
+        soft_assert(c_packages == e_packages, "packages: '{}' != '{}'".format(c_packages,
+                                                                              e_packages))
+        soft_assert(c_services == e_services,
+                    "services: '{}' != '{}'".format(c_services, e_services))
     else:
         # Make sure windows-specific data is not empty
         c_patches = InfoBlock.text('Security', 'Patches')

--- a/cfme/tests/infrastructure/test_instance_analysis.py
+++ b/cfme/tests/infrastructure/test_instance_analysis.py
@@ -22,8 +22,7 @@ from utils.log import logger
 from utils.wait import wait_for
 from utils.blockers import BZ
 
-pytestmark = [pytest.mark.tier(3), test_requirements.smartstate,
-              pytest.mark.meta(blockers=[1378447])]
+pytestmark = [pytest.mark.tier(3), test_requirements.smartstate]
 
 WINDOWS = {'id': "Red Hat Enterprise Windows", 'icon': 'windows'}
 
@@ -182,32 +181,23 @@ def vm_analysis_data(provider, analysis_type):
     pdata = provider.data
     provisioning_data = pdata.get('vm_analysis_new', {}).get('provisioning', {})
 
-    # Setup up the provisioning data for the instance/vm
+    # Setup the provisioning data for the instance/vm
     # Will default to provisioning items under vm_analysis but if not defined
     #   falls back to items under provider['provisioning'] key
-    if 'host' not in provisioning_data and 'host' in pdata['provisioning']:
-        provisioning_data['host'] = pdata['provisioning']['host']
-    if 'datastore' not in provisioning_data and 'datastore' in pdata['provisioning']:
-        provisioning_data['datastore'] = pdata['provisioning']['datastore']
-    if 'vlan' not in provisioning_data and 'vlan' in pdata['provisioning']:
-        provisioning_data['vlan'] = pdata['provisioning']['vlan']
-    if 'instance_type' not in provisioning_data and 'instance_type' in pdata['provisioning']:
-        provisioning_data['instance_type'] = pdata['provisioning']['instance_type']
-    if 'availability_zone' not in provisioning_data and \
-            'availability_zone' in pdata['provisioning']:
-        provisioning_data['availability_zone'] = pdata['provisioning']['availability_zone']
-    if 'security_group' not in provisioning_data and 'security_group' in pdata['provisioning']:
-        provisioning_data['security_group'] = pdata['provisioning']['security_group']
-    if 'cloud_network' not in provisioning_data and 'cloud_network' in pdata['provisioning']:
-        provisioning_data['cloud_network'] = pdata['provisioning']['cloud_network']
+    provisioning_data.setdefault('host', pdata['provisioning']['host'])
+    provisioning_data.setdefault('datastore', pdata['provisioning']['datastore'])
+    provisioning_data.setdefault('vlan', pdata['provisioning']['vlan'])
+    provisioning_data.setdefault('instance_type', pdata['provisioning']['instance_type'])
+    provisioning_data.setdefault('availability_zone', pdata['provisioning']['availability_zone'])
+    provisioning_data.setdefault('security_group', pdata['provisioning']['security_group'])
+    provisioning_data.setdefault('cloud_network', pdata['provisioning']['cloud_network'])
 
     # If defined, tries to find cluster from provisioning, then provider definition itself
     if provider.type == 'rhevm':
-        if 'cluster' not in provisioning_data:
-            if 'cluster' not in pdata['provisioning']:
-                provisioning_data['cluster'] = pdata['default_cluster']
-            else:
-                provisioning_data['cluster'] = pdata['provisioning']['cluster']
+        if 'cluster' not in provisioning_data and 'cluster' not in pdata['provisioning']:
+            provisioning_data['cluster'] = pdata['default_cluster']
+        else:
+            provisioning_data['cluster'] = pdata['provisioning']['cluster']
 
     provisioning_data.update(
         provider.data.get('vm_analysis_new', {}).get('vms', {}).get(analysis_type, {}))
@@ -419,7 +409,6 @@ def test_ssa_vm(provider, instance, soft_assert):
     #       delete the vm itself if it did have a scan already
     #       delete all previous scan tasks
 
-
     e_users = None
     e_groups = None
     e_packages = None
@@ -488,6 +477,7 @@ def test_ssa_vm(provider, instance, soft_assert):
         soft_assert(c_kernel_drivers != '0', "kernel drivers: '{}' != '0'".format(c_kernel_drivers))
         soft_assert(c_fs_drivers != '0', "fs drivers: '{}' != '0'".format(c_fs_drivers))
 
+    # TODO: revisit this and see if we should re-enable it
     # image_label = 'Parent VM'
     # if provider.type == 'openstack':
     #     image_label = 'VM Template'

--- a/cfme/tests/infrastructure/test_instance_analysis.py
+++ b/cfme/tests/infrastructure/test_instance_analysis.py
@@ -7,6 +7,7 @@ import pytest
 from cfme import test_requirements
 from cfme.common.vm import VM, Template
 from cfme.common.provider import cleanup_vm
+from cfme.cloud.provider import CloudProvider
 from cfme.configure import configuration
 from cfme.configure.tasks import is_vm_analysis_finished
 from cfme.control.explorer.policy_profiles import PolicyProfile
@@ -184,13 +185,16 @@ def vm_analysis_data(provider, analysis_type):
     # Setup the provisioning data for the instance/vm
     # Will default to provisioning items under vm_analysis but if not defined
     #   falls back to items under provider['provisioning'] key
-    provisioning_data.setdefault('host', pdata['provisioning']['host'])
-    provisioning_data.setdefault('datastore', pdata['provisioning']['datastore'])
-    provisioning_data.setdefault('vlan', pdata['provisioning']['vlan'])
-    provisioning_data.setdefault('instance_type', pdata['provisioning']['instance_type'])
-    provisioning_data.setdefault('availability_zone', pdata['provisioning']['availability_zone'])
-    provisioning_data.setdefault('security_group', pdata['provisioning']['security_group'])
-    provisioning_data.setdefault('cloud_network', pdata['provisioning']['cloud_network'])
+
+    if not isinstance(provider, CloudProvider):
+        provisioning_data.setdefault('host', pdata['provisioning']['host'])
+        provisioning_data.setdefault('datastore', pdata['provisioning']['datastore'])
+        provisioning_data.setdefault('vlan', pdata['provisioning']['vlan'])
+    if isinstance(provider, CloudProvider):
+        provisioning_data.setdefault('instance_type', pdata['provisioning']['instance_type'])
+        provisioning_data.setdefault('availability_zone', pdata['provisioning']['availability_zone'])
+        provisioning_data.setdefault('security_group', pdata['provisioning']['security_group'])
+        provisioning_data.setdefault('cloud_network', pdata['provisioning']['cloud_network'])
 
     # If defined, tries to find cluster from provisioning, then provider definition itself
     if provider.type == 'rhevm':

--- a/utils/virtual_machines.py
+++ b/utils/virtual_machines.py
@@ -32,13 +32,13 @@ def _vm_cleanup(mgmt, vm_name):
             'Could not destroy VM/instance %s (%s: %s)', vm_name, type(f).__name__, str(f))
 
 
-def deploy_template(provider_key, vm_name, template_name=None, timeout=900,
-        **deploy_args):
+def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **deploy_args):
     """
     Args:
         provider_key: Provider key on which the VM is to be created
         vm_name: Name of the VM to be deployed
         template_name: Name of the template that the VM is deployed from
+        timeout: the timeout for template deploy
     """
     allow_skip = deploy_args.pop("allow_skip", ())
     if isinstance(allow_skip, dict):

